src/stores/sdkStore.ts

```diff
@@ -0,0 +1,200 @@
+
+  async function scanForWallets() {
+    status.value = Status.WALLET_SCANNING;
+
+    return new Promise((resolve, reject) => {
+      if (aeSdk instanceof AeSdkAepp) {
+        const handleWallets: ({
+          wallets,
+          newWallet,
+        }: {
+          wallets: Wallets;
+          newWallet?: Wallet | undefined;
+        }) => Promise<void> = async ({ wallets, newWallet }) => {
+          newWallet = newWallet || Object.values(wallets)[0];
+          stopScan();
+          if (!aeSdk) return;
+
+          try {
+            (aeSdk as AeSdkAepp).disconnectWallet();
+          } catch (_) {}
+
+          const { networkId } = await (aeSdk as AeSdkAepp).connectToWallet(
+            newWallet.getConnection()
+          );
+
+          aeSdk.selectNode(networkId);
+
+          await (aeSdk as AeSdkAepp).subscribeAddress(
+            SUBSCRIPTION_TYPES.subscribe,
+            "current"
+          );
+
+          resolve("");
+        };
+        const scannerConnection = new BrowserWindowMessageConnection();
+        const stopScan = walletDetector(scannerConnection, handleWallets);
+      } else reject("AeSdk instance needed for wallet connection");
+    });
+  }
+
+  async function initAeSdkAepp(customNodeUrl?: string) {
+    isStatic.value = false;
+    aeSdk = new AeSdkAepp({
+      name: "Contract Editor",
+      ...getNodes(customNodeUrl),
+      onCompiler: new CompilerHttp(COMPILER_URL, { ignoreVersion: true }),
+      onNetworkChange: async ({ networkId }) => {
+        if (aeSdk) {
+          const [{ name }] = (await aeSdk.getNodesInPool()).filter(
+            (node) => node.nodeNetworkId === networkId
+          );
+          aeSdk.selectNode(name);
+          await updateConnectionInfo();
+        }
+      },
+      onAddressChange: () => updateConnectionInfo(),
+      onDisconnect: () => alert("Aepp is disconnected"),
+    });
+
+    await scanForWallets();
+  }
+
+  function initAeSdk(secretKey: string, customNodeUrl?: string) {
+    isStatic.value = true;
+    aeSdk = new AeSdk({
+      onCompiler: new CompilerHttp(COMPILER_URL, { ignoreVersion: true }),
+      accounts: [new MemoryAccount(secretKey)],
+      ...getNodes(customNodeUrl),
+    });
+  }
+
+  async function initSdk(initStatic: boolean = true, customNodeUrl?: string) {
+    status.value = Status.UNINITIALIZED;
+    if (customNodeUrl) {
+      nodeUrl.value = customNodeUrl;
+    } else {
+      nodeUrl.value = nodes[0].instance.$host;
+    }
+
+    try {
+      if (initStatic) {
+        secretKey.value = getOrGenerateSecretKey();
+
+        if (secretKey.value) initAeSdk(secretKey.value, customNodeUrl);
+      } else await initAeSdkAepp(customNodeUrl);
+
+      await updateConnectionInfo();
+    } catch (e) {
+      status.value = Status.CONNECTION_ERROR;
+    }
+  }
+
+  async function updateConnectionInfo() {
+    status.value = Status.FETCHING_INFO;
+    networkId.value = await aeSdk?.api?.getNetworkId();
```

Comment:
should we reject here when aeSdk is not defined?

Reply:
didn't see a good way to do it and currently the case won't happen as far as I know